### ISSUE TITLE
Expand list of approvers and set up auto-merge for org-management PRs

### ISSUE THREAD

**Issue by accorvin** [opendatahub-io/org-management#24]
Right now, we require manual approval and merging from the authorized set of owners for this org. This introduces a bottleneck on adding new people to the org.

I believe we should:
1) Increase the number of people listed as approvers for PRs to this repo
2) Implement automation such that when a PR is approved by an authorized approver, the PR is automatically merged and org membership changes are applied.

**Comment by accorvin** [comment]
@jkoehler-redhat @LaVLaS @VaishnaviHire what do you think of this proposal. If you approve, I can volunteer my team's time to make it happen.

cc @anishasthana

**Comment by LaVLaS** [comment]
> Increase the number of people listed as approvers for PRs to this repo
> Implement automation such that when a PR is approved by an authorized approver, the PR is automatically merged and org membership changes are applied.
I agree with both points.  @accorvin I have some basic automation in development but if your team is will to create and own the automation then it would be greatly appreciated

> @accorvin I think this is fine as long as all approvers know to approve aligned to the community guidelines: https://github.com/opendatahub-io/opendatahub-community/blob/main/community-membership.md

Since the previous updates have been blind invites with minimal context , I added templates to  outline the requirements for requesting organization membership for all new PRs.

**Comment by anishasthana** [comment]
So will we require an issue for every new membership PR? Or is just a PR following the template sufficient?

**Comment by anishasthana** [comment]
I can spend some time working on a github action automating it next week. I'm thinking for now we can start with the same OWNERS list  + myself and Alex. We can iterate over time to better tie in SIG leadership

**Comment by LaVLaS** [comment]
> For members of the RHODS team (including dev/qe/docs/program/sre/bu/etc), the issue template feels to heavy weight to me. We should be able to go straight to PR.

This is a good point we were discussing in the steering committe.  The original goal was to have a self governing community so the idea that there is a RHODS "team" controlling the community goes against the idea of community governance.  If we are auto-adding internal RHODS team members or RH OpenShift AI contributors then we just state that the membership requirements are you have to be an IBM/Red Hat employee or partners

**Comment by accorvin** [comment]
This is handled by #48

**Comment by grdryn** [comment]
Howdy, I'm new _here_ :slightly_smiling_face: 

I'm an admin of an org-management/Peribolos config for a different org here, and there might be some automation that I could propose in a PR here that you might like:

- We do a Peribolos dry run on each PR automatically (typically takes about ~1min)
- We do a Peribolos run (not dry) on each merge (similarly takes 60-90s)
- We schedule that same Peribolos run daily, to flush out any manual changes and force people to remember that they need to make their changes here through the config :)

Currently I see that for this org, the Peribolos run appears to be triggered manually by the person who merges the PR, and it typically takes more than 20 minutes (and it seem to do much less / is a much smaller config than the other org that I'm talking about), so I think what I'm suggesting would be a positive improvement.

(I haven't looked into why the other one is so quick compared to this one, so I'm not sure)

Let me know what you think, thanks! :slightly_smiling_face:

**Comment by anishasthana** [comment]
The source for that image is in https://github.com/opendatahub-io/org-management/tree/main/.github/build.

@accorvin  recently created https://github.com/opendatahub-io/org-management/blob/main/.github/workflows/peribolos.yaml previously, so in theory any time a PR is merged in the membership is updated.

That being said, please raise PRs for anything you think you can improve. I'm happy to chat synchronously/asynchronously as well if you need someone to bounce ideas off

**Comment by grdryn** [comment]
Thanks for the info @anishasthana, I'll try to find some time to look at it soon and see if there are improvements that I can suggest in a PR :+1:

**Comment by accorvin** [comment]
@grdryn just chiming in to say that we strongly welcome any automation improvements you're willing to contribute to this repo!
